src/libsystemd/sd-event/sd-event.c

```diff
@@ -3227,6 +3231,152 @@ static int process_inotify(sd_event *e) {
         return done;
 }
 
+static void source_ratelimit_state_restore(sd_event_source *s) {
+        assert(s);
+
+        s->type = s->ratelimit_saved_state.type;
+        s->wakeup = s->ratelimit_saved_state.wakeup;
+        s->prepare = s->ratelimit_saved_state.prepare;
+        s->destroy_callback = s->ratelimit_saved_state.destroy_callback;
+        s->userdata = s->ratelimit_saved_state.userdata;
+
+        memcpy(&s->io, s->ratelimit_saved_state.data, s->ratelimit_saved_state.size);
+
+        s->enabled ? (void) source_enable(s, s->enabled) : (void) source_disable(s);
```

Comment:
hmm, this ignores errors. ignoring them for source_disable() is probably OK, it cannot realistically fail. But I am not happy about ignoring source_enable failures... there needs to be some error handling there.